mx.py

```diff
@@ -1893,11 +1905,84 @@ def buildJavacLike(self, jdk, project, jvmArgs, javacArgs, disableApiRestriction
             abort('Showing task tags is not currently supported for javac')
         javacArgs.append('-encoding')
         javacArgs.append('UTF-8')
-        javac = self.altJavac if self.altJavac else jdk.javac
         jvmArgs += jdk.java_args
+        self.run(jdk, jvmArgs, javacArgs)
+
+    def run(self, jdk, jvmArgs, javacArgs):
+        javac = self.altJavac if self.altJavac else jdk.javac
         cmd = [javac] + ['-J' + arg for arg in jvmArgs] + javacArgs
         run(cmd)
 
+class JavacCompilerDaemon(JavacCompiler):
+    def __init__(self, altJavac=None, extraJavacArgs=None):
+        JavacCompiler.__init__(self, altJavac, extraJavacArgs)
+
+    def name(self):
+        return 'javac-daemon'
+
+    def run(self, jdk, jvmArgs, javacArgs):
+        # ignore the jvmArgs
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.connect(('127.0.0.1', self.daemon.port))
+        s.send(" ".join(javacArgs) + "\n")
+        f = s.makefile()
+        retcode = int(f.readline())
+        s.close()
+        nonZeroIsFatal = True
+        if retcode and nonZeroIsFatal:
+            if _opts.verbose:
+                if _opts.very_verbose:
+                    raise subprocess.CalledProcessError(retcode, jdk.javac + ' '.join(javacArgs))
+                else:
+                    log('[exit code: ' + str(retcode) + ']')
+            abort(retcode)
+
+        return retcode
+
+    def prepare(self, jdk, daemons):
+        name = 'javac-daemon-' + jdk.java
+        self.daemon = daemons.get(name)
+        if not self.daemon:
+            self.daemon = JavacDaemon(jdk)
+            daemons[name] = self.daemon
+
+class Daemon:
+    def shutdown(self):
+        pass
+
+class JavacDaemon(Daemon):
+    def __init__(self, jdk):
+        self.jdk = jdk
+        self.mainClass = 'com.oracle.mxtool.compilerserver.JavacDaemon'
+        if not exists(join(project('com.oracle.mxtool.compilerserver').output_dir(), self.mainClass.replace('.', os.sep) + '.class')):
+            build(['--only', 'com.oracle.mxtool.compilerserver', '--no-javac-daemon'])
```

Comment:
I'll switch to --dependencies and make it unconditional.  I was just following what junit was doing since I didn't know what the right way to handle this was.